src/read.rs

```diff
@@ -226,48 +225,45 @@ impl<R> Reader<R>
     /// The format is determined by attempting to construct each specific format reader until one
     /// is successful.
     pub fn new(mut reader: R) -> Result<Self, ReadError> {
-        let mut reader_errors = Vec::new();
 
         #[cfg(feature="wav")]
         {
-            let maybe_err = match hound::WavReader::new(&mut reader) {
-                Ok(_) => None,
-                Err(err) => Some(err),
+            let is_wav = match hound::WavReader::new(&mut reader) {
+                Err(hound::Error::FormatError(_)) => false,
+                _ => true,
```

Comment:
Oh right, we construct them twice.... Maybe it would be a good idea to re-use it? E.g. if its Ok(s) we return `Reader::Wav(s)` and be happy, and otherwise we either propagate the error (e.g. for io errors), or we continue (if the error was about the wrong format).

Just an anecdote, I had some errors in my HDD, and every time I wanted to read a specific file, it first hanged for some seconds and then errored. If we ignore IO errors and retry multiple times with different formats this means a much longer period to wait.